compiler/rustc_hir_analysis/src/check/compare_method.rs

```diff
@@ -628,15 +626,15 @@ struct ImplTraitInTraitCollector<'a, 'tcx> {
     types: FxHashMap<DefId, (Ty<'tcx>, ty::SubstsRef<'tcx>)>,
     span: Span,
     param_env: ty::ParamEnv<'tcx>,
-    body_id: hir::HirId,
+    body_id: hir::def_id::LocalDefId,
```

Comment:
nit: add `use rustc_span::def_id::LocalDefId;` to the top of this function instead.